### ISSUE TITLE
False Positive: "react/prop-types [PROP_NAME] is missing in props validation" when using stateless components + destructured param + Flow

### ISSUE THREAD

**Issue by ayrton** [yannickcr/eslint-plugin-react#498]
Similar to #467, I found a minor bug with stateless components with flow annotations.

```sh
/Users/adc/Developer/ayrton/react-key-handler/demo/components/demos/input-element.js
  9:16  error  'keyName' is missing in props validation  react/prop-types
```

```js
/* @flow */

import React from 'react';

type Props = {
  keyName: ?string,
};

function Demo({keyName}: Props): React$Element {
  return (
    <div>
      {keyName}
    </div>
  );
}
```

This is caused because of destructuring, when I don't destructure the props all is fine:

```js
function Demo(props: Props): React$Element {
  return (
    <div>
      {props.keyName}
    </div>
  );
}
```

**Comment by shrijan00003** [comment]
is there any update or solution for this issue ??

**Comment by ljharb** [comment]
It’s been fixed for years. Please file a new issue if you’re having trouble.

**Comment by alekpentchev** [comment]
Hi all. I have this issue when I'm using stateless functional components and Flow. The code looks like this:
```
const renderForm = (): Node => {
    const { action } = props;
    const { form } = css;

    if (!action) {
      return null;
    }
    return (
      <form
        onSubmit={onSubmit}
        acceptCharset="UTF-8"
        className={form}
        method="post"
        action={action}
      >
        {displayInputs()}
      </form>
    );
  };
```

several lines above there is component declaration:
```
export function Form(props: Props) {
...
}
```

Props are imported from other file.

My eslint-plugin-react version is 7.14.3. So how should I solve this problem?

**Comment by ljharb** [comment]
renderForm should be a component, but perhaps Form needs PropTypes. It’s hard to tell. Can you file a new issue, after upgrading to the latest version of this plugin?

**Comment by alekpentchev** [comment]
> renderForm should be a component, but perhaps Form needs PropTypes. It’s hard to tell. Can you file a new issue, after upgrading to the latest version of this plugin?

Yes, apparently. After adding PropTypes this error is no longer reported by eslint. Which is strange. And, in my opinion, it's not the best approach to have Flow and PropTypes for defining types.

**Comment by ljharb** [comment]
Both are required; neither flow nor typescript is capable of catching everything PropTypes can.

**Comment by ExpectoPatrom** [comment]
I'm using eslint-plugin-react 7.20.3 and I'm having the same issue, do you have any idea of what should do?

**Comment by ljharb** [comment]
@ExpectoPatrom would you mind filing a new issue? please include the code and the warnings.

**Comment by huntu1992** [comment]
@ExpectoPatrom same here

**Comment by matthewgoslett** [comment]
I can confirm this is happening for me on ^7.21.5

**Comment by ExpectoPatrom** [comment]
Hey, nope. It seems like this is not a bug, instead it's kind a bad practice and there's another way of doing the components. I can't remember it specifically but I'm gonna search and then I post it

**Comment by alielkhateeb** [comment]
This is happening to me whether I destruct or not, is anyone else facing this issue?

**Comment by ljharb** [comment]
@alielkhateeb please file a new issue

**Comment by ttcode10** [comment]
I found this is useful: [https://hashnode.blainegarrett.com/making-eslint-happy-in-mixed-typescriptjavascript-projects-ck5lge2v204cgqks1sk4nlp85](url)

**Comment by denven** [comment]
eslint-plugin-react": "^7.23.1",

```js
AppProvider.propTypes = {
  children: PropTypes.any.isRequired,
};
```
```
  21:8   error  'React' is defined but never used          no-unused-vars  # No React imported in the js file
  74:24  error  'children' is missing in props validation  react/prop-types # Still complains
```

**Comment by ljharb** [comment]
@denven please file a new issue.

**Comment by Dinkelborg** [comment]
How is this still an issue after 5 years? 

`interface Props {
  children?: React.ReactNode;
}

const SomeHandlerComponent: FC<Props> = ({ children }) => {`

ESLint 2.1.23

**Comment by mosoakinyemi** [comment]
This is still an issue?
Guys kindly look at this 🔧

**Comment by PerfectionVR** [comment]
this will fail, complaining that text, callback isn't prop validated, this resulted in
```
  21:46  error  'text' is missing in props validation      react/prop-types
  21:52  error  'callback' is missing in props validation  react/prop-types
```
```ts
const Markdown: React.FC<MarkdownProps> = ({ text, callback }: MarkdownProps): React.ReactElement => {
  const settings = {
    createElement: React.createElement,
    components: {},
    elements: {
      p(props) {
      // return some component
      },
    },
  };
  const compile = marksy(settings);
  const description = compile(text).tree;
  return <div className="markdown">{description}</div>;
};
```

this will be accepted as valid though
```ts
cont p: React.FC = ()=>{};
const Markdown: React.FC<MarkdownProps> = ({ text, callback }: MarkdownProps): React.ReactElement => {
  const settings = {
    createElement: React.createElement,
    components: {},
    elements: {
      p,
    },
  };
  const compile = marksy(settings);
  const description = compile(text).tree;
  return <div className="markdown">{description}</div>;
};
```

I was able to repeatedly repoduce between these two scenarios

**Comment by s4shebi** [comment]
one can turn off this type of errors by just adding `"react/prop-types": ["off"],` in the rules of `.eslintrc.js`

**Comment by ideepakpg** [comment]
any solution for " 'prop' is mssing in prop validation " error except disabling prop types

**Comment by ljharb** [comment]
@ideepakpg please file a new issue and i'll take a look
